internal/controller/kustomization_controller.go

```diff
@@ -181,6 +181,11 @@ func (r *KustomizationReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Finalise the reconciliation and report the results.
 	defer func() {
+		// Do not proceed if the Kustomization is suspended
+		if obj.Spec.Suspend {
+			return
+		}
+
```

Comment:
This breaks the finalization and resources. Please move this just before `if conditions.IsReady(obj) {`.

Reply:
Thanks for the feedback. I updated the commit accordingly.